share/man/man9/cdefs.9

```diff
@@ -0,0 +1,297 @@
+.It Sy __GLOBAL(sym) Ta
+.It Sy __WEAK(sym) Ta
+.It Sy __DECONST(type, var) Ta
+.It Sy __DEVOLATILE(type, var) Ta
+.It Sy __DEQUALIFY(type, var) Ta
+.It Sy __RENAME(x) Ta
+.El 
+.Ss Emulated Keywords
+As C evolves, many of the old macros we once used have been incorporated into
+the standard language.
+As this happens, we add support for these keywords as macros for older
+compilation environments.
+Sometimes this results in a nop in the older environment.
+.Bl -column ".Sy Keyword" ".Sy Description"
+.It Sy Keyword Ta Sy Description
+.It Sy _Alignas(x) Ta
+.It Sy _Alignof(x) Ta
+.It Sy _Noreturn Ta Expandsd to
+.Dq [[noreturn]]
+in C++-11 and newer compilation environments, otherwise
+.Dq __dead2
+.It _Static_assert(x, y) Ta Compile time assertion that
+.Fa x
+is true, otherwise emit
+.Fa y
+as the error message.
+.It Sy _Thread_local Ta Designate variable as thread local storage.
+.It Sy __generic Ta implement _Generic-like features which aren't entirely possible to emulate the _Generic keyword.
+.It Sy __noexcept Ta to emulate the C++11 argument-less noexcept form.
+.It Sy __noexcept_if Ta to emulate the C++11 conditional noexcept form.
+.El
+.Ss Support Macros
+The following macros are defined, or have specific values, to denote certain
+things about the build environment.
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Sy __LONG_LONG_SUPPORTED Ta Variables may be declared
+.Dq long long .
+This is defined for C99 or newer and C++ environments.
+.It Sy __STDC_LIMIT_MACROS Ta
+.It Sy __STDC_CONSTANT_MACROS Ta
+.El
+.Ss Nice to have macros
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Sy __offsetof(type,field) Ta
+.It Sy __rangeof(type,start,end) Ta
+.It Sy __containerof(x,s,m) Ta
+.El
+.Ss ID Strings
+This section is deprecated, but is kept around because too much contrib software
+still uses these.
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Sy __IDSTRING(name,string) Ta
+.It Sy __FBSDID(s) Ta
+.It Sy __RCSID(s) Ta
+.It Sy __RCSID_SOURCE(s) Ta
+.It Sy __SCCSID(s) Ta
+.It Sy __COPYRIGHT(s) Ta
+.El
+.Sh Supported C Environments
+FreeBSD supports a number C standard environments.
+Selection of the language dialect is a compiler dependent command line option,
+though it is usually
+.Fl cstd=XX
+where XX is the standard you desire, like c89 or c23.
+FreeBSD provides a number of selection macros to control visibility of symbols.
+Please see the section on Selection Macros for the specifics.
+.Bl -tag -compact
+.It K \*(Am R
+Pre-ansi Kernighan and Ritchie C.
+Sometimes called
+.Dq knr
+or
+.Dq C78
+to distinguish it from newer standards.
+Support for this compilation environment is dependent on compilers supporting
+this configuration.
+Most of the old forms of C have been deprecated or removed in
+ISO/IEC 9899:2024 (“ISO C23”) and newer compilers may require many warnings to
+be suppressed to active this mode.
```

Comment:
```suggestion
ISO/IEC 9899:2024 (“ISO C23”).
Newer compilers may require suppression of many warnings to activate this mode.
```

_activate_ (not _active_)

Reply:
ah, yes. I've taken a slightly different rewording, please feel free to comment on it for the next round.